demos/twitter/src/main/java/com/datatorrent/demos/twitter/TwitterURLExtractorWithFilter.java

```
@@ -0,0 +1,108 @@
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.common.util.BaseOperator;
+
+import twitter4j.Status;
+import twitter4j.URLEntity;
+
+/**
+ * <p>
+ * TwitterStatusURLExtractorWithFilter class.
+ * </p>
+ *
+ * @since 0.3.2
+ */
+public class TwitterURLExtractorWithFilter extends BaseOperator
+{
+
+  protected List<String> myList = new ArrayList<String>();
+  BufferedReader in;
+
+  public final transient DefaultOutputPort<String> url = new DefaultOutputPort<String>();
+  public final transient DefaultInputPort<Status> input = new DefaultInputPort<Status>()
+  {
+    @Override
+    public void process(Status status)
+    {
+      if (myList.isEmpty()) {
+        try {
+          DocumentBuilderFactory dbFactory = DocumentBuilderFactory
+              .newInstance();
+          DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+          InputStream is = this.getClass().getClassLoader()
+              .getResourceAsStream("domainlist.xml");
+          Document doc = dBuilder.parse(is);
+          doc.getDocumentElement().normalize();
+          NodeList nList = doc.getElementsByTagName("sub");
+
+          for (int temp = 0; temp < nList.getLength(); temp++) {
+            Node nNode = nList.item(temp);
+            if (nNode.getNodeType() == Node.ELEMENT_NODE) {
+              Element eElement = (Element) nNode;
+              String items = eElement.getElementsByTagName("value").item(0)
+                  .getTextContent();
+              myList = Arrays.asList(items.split(", "));
+            }
+          }
+
+        } catch (ParserConfigurationException e) {
+          throw new RuntimeException("Parser Configuration Exception");
```

Please don't mask real exception.